### PR TITLE
Initial DO impl commit

### PR DESCRIPTION
**Pull request by kt5356**
Doing an initial commit to get the basic structure committed.

Added new Byoc base client to combine some keys and functions that will be common to current and future clouds.

### CODE REVIEW

**Review by lionello — APPROVED** [review]
LGTM! but I'd like @edwardrf  to look at it too

File: src/pkg/clouds/do/common.go
Comment (edwardrf) [inline src/pkg/clouds/do/common.go:1]: I think we might not need this file anymore, probably we can merge it with `do/region/region.go`

File: src/pkg/cli/client/byoc/common.go
Comment (edwardrf) [inline src/pkg/cli/client/byoc/common.go:1]: i'll prefer rename this file to baseclient.go

**Review by edwardrf — APPROVED** [review]
LGTM

File: src/pkg/clouds/do/common.go
Comment (kt5356) [reply src/pkg/clouds/do/common.go:1]: k, will see what I can do in the next iteration.

File: src/pkg/cli/client/byoc/common.go
Comment (kt5356) [reply src/pkg/cli/client/byoc/common.go:1]: renamed

**Pull request merged**